spec/features/cases/offender_sar/add_data_requests_spec.rb

```diff
@@ -51,6 +51,48 @@
     expect(last_row.total_value).to have_text '0'
   end
 
+  scenario 'total page count is correct', js: true do
+    cases_show_page.load(id: offender_sar_case.id)
+    click_on 'Record data request'
+    expect(data_request_page).to be_displayed
+
+    request_values = [
+        { location: 'HMP Leicester  ', data: ' A list of meals fed to mo  '},
+        { location: 'HMP Brixton', data: 'Latest menu from the clink by chef mo'},
+    ]
+
+    request_values.each_with_index do |request, i|
+      data_request_page.form.location[i].fill_in(with: request[:location])
+      data_request_page.form.data[i].fill_in(with: request[:data])
+    end
+
+    click_on 'Record requests'
+
+    cases_show_page.data_requests.rows[0].click_on 'Update page count'
+
+    data_request_edit_page.form.date_received_dd.fill_in(with: 11)
+    data_request_edit_page.form.date_received_mm.fill_in(with: 6)
+    data_request_edit_page.form.date_received_yyyy.fill_in(with: 2020)
+    data_request_edit_page.form.num_pages.fill_in(with: 32)
+
+    click_on 'Update data received'
+
+    last_row = cases_show_page.data_requests.rows.last
+    expect(last_row.total_value).to have_text '32'
+
+    cases_show_page.data_requests.rows[1].click_on 'Update page count'
+
+    data_request_edit_page.form.date_received_dd.fill_in(with: 11)
+    data_request_edit_page.form.date_received_mm.fill_in(with: 6)
+    data_request_edit_page.form.date_received_yyyy.fill_in(with: 2020)
+    data_request_edit_page.form.num_pages.fill_in(with: 128)
+
+    click_on 'Update data received'
+
+    last_row = cases_show_page.data_requests.rows.last
+    expect(last_row.total_value).to have_text '160'
+  end
+
```

Comment:
Thanks for adding the test! You've done the right thing in terms of a feature spec, clicking buttons in the UI to add the data, I just wonder whether adding another scenario and adding data requests manually (which is already covered in a feature spec) might be a little heavy handed in this case, when all we want to do is verify that the total calculation of that one line is correct. Ideally the place to test that would be in a view spec, but those can be fiddly and because you need to supply all the state they need to work, they're a bit artificial. But for a calculation I'd probably do it.  

For this tiny feature I think it's fine and pragmatic to use the existing feature spec, but maybe consider creating test data in the state you want and just verify the page displays with the total correct:

```suggestion

  context 'when multiple data requests are present and have pages logged' do
    given!(:data_request) { create(:data_request, offender_sar_case: offender_sar_case, cached_num_pages: 32) }
    given!(:second_data_request) { create(:data_request, offender_sar_case: offender_sar_case, cached_num_pages: 32) }

    scenario 'the total row displays with the correct total pages' do
      cases_show_page.load(id: offender_sar_case.id)
      expect(cases_show_page).to be_displayed
      expect(cases_show_page.data_requests.rows.size).to eq 3 # Unchanged num DataRequest

      last_row = cases_show_page.data_requests.rows.last
      expect(last_row).to have_selector '.total-label'
      expect(last_row.total_label).to have_text 'Total'
      expect(last_row).to have_selector '.total-value'
      expect(last_row.total_value).to have_text '64'
    end
  end
```

Reply:
I've also moved this test to the log_data_recieved_spec.rb file.  Think thats the better context for it.